src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ExplicitConversionSymbolReferenceFinder.cs

```diff
@@ -47,16 +47,11 @@ protected override bool CanFind(IMethodSymbol symbol)
         Contract.ThrowIfNull(underlyingNamedType);
 
         using var _ = PooledHashSet<Document>.GetInstance(out var result);
-        await FindDocumentsAsync(project, documents, StandardCallbacks<Document>.AddToHashSet, result, cancellationToken, underlyingNamedType.Name).ConfigureAwait(false);
-        await FindDocumentsAsync(project, documents, underlyingNamedType.SpecialType.ToPredefinedType(), StandardCallbacks<Document>.AddToHashSet, result, cancellationToken).ConfigureAwait(false);
+        await FindDocumentsAsync(searchEngine, project, documents, StandardCallbacks<Document>.AddToHashSet, result, cancellationToken, underlyingNamedType.Name).ConfigureAwait(false);
+        await FindDocumentsAsync(searchEngine, project, documents, underlyingNamedType.SpecialType.ToPredefinedType(), StandardCallbacks<Document>.AddToHashSet, result, cancellationToken).ConfigureAwait(false);
 
         // Ignore any documents that don't also have an explicit cast in them.
-        foreach (var document in result)
-        {
-            var index = await SyntaxTreeIndex.GetRequiredIndexAsync(document, cancellationToken).ConfigureAwait(false);
-            if (index.ContainsConversion)
-                processResult(document, processResultData);
-        }
+        await FindDocumentsWithPredicateAsync(searchEngine, project, documents, index => index.ContainsConversion, processResult, processResultData, cancellationToken).ConfigureAwait(false);
```

Comment:
the above loop is what FindDocumentsWithPredicateAsync already does.  so converted to that.